pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -183,6 +194,8 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, pr *v1beta1.PipelineRun)
 		logger.Errorf("Failed to fetch pipeline func for pipeline %s: %w", pr.Spec.PipelineRef.Name, err)
 		pr.Status.MarkFailed(ReasonCouldntGetPipeline, "Error retrieving pipeline for pipelinerun %s/%s: %s",
 			pr.Namespace, pr.Name, err)
+		span.SetStatus(codes.Error, "Failed to fetch pipelinefunc")
```

Comment:
NIT: Perhaps "Failed to fetch the pipeline definition" - and include the error too?

Reply:
error is recorded in the line below using `span.RecordError(err)`